hedvalidation/hedvalidation/hed_input_reader.py

```diff
@@ -72,9 +72,11 @@ def __init__(self, hed_input, tag_columns=[2], has_column_names=True, check_for_
         self._tag_columns = self._convert_tag_columns_to_processing_format(tag_columns);
         self._has_column_names = has_column_names;
         self._worksheet_name = worksheet_name;
-        self._hed_dictionary = self._get_hed_dictionary(hed_xml_file);
-        self._tag_validator = TagValidator(self._hed_dictionary, check_for_warnings=check_for_warnings);
+        if run_semantic_validation:
+            self._hed_dictionary = self._get_hed_dictionary(hed_xml_file);
+        self._tag_validator = TagValidator(self._hed_dictionary, check_for_warnings=check_for_warnings, run_semantic_validation=run_semantic_validation);
```

Comment:
`self._hed_dictionary` is undefined if semantic validation is off.

Reply:
should the if statement be removed then?